python/src/equistore/operations/subtract.py

```diff
@@ -30,15 +31,9 @@ def subtract(A: TensorMap, B: Union[float, TensorMap]) -> TensorMap:
 
     :return: New :py:class:`TensorMap` with the same metadata as ``A``.
     """
-
     if isinstance(B, TensorMap):
         _check_maps(A, B, "subtract")
-        for key, blockB in B:
-            B.block(key).values[:] = -1 * B.block(key).values[:]
-            for parameter in blockB.gradients_list():
-                B.block(key).gradient(parameter).data[:] = (
-                    -B.block(key).gradient(parameter).data[:]
-                )
+        B = multiply(A=B, B=-1)
```

Comment:
```suggestion
        B = multiply(B, -1)
```

Reply:
So this is fine for now, but also not the best solution: we are still creating a new tensormap to hold `-B`, but we don't have to. I'll open an issue to improve later